posthog/hogql/printer.py

```diff
@@ -107,6 +107,7 @@ def __init__(
         self.dialect = dialect
         self.stack: List[AST] = stack or []  # Keep track of all traversed nodes.
         self.settings = settings
+        self.in_join_on = False
```

Comment:
This is _super_ bespoke. What if we made `visitJoinConstraintClause` return an `ast.JoinConstraint` and just used the stack for this? (There's some TODO in another file, but I'm not sure if it's about this exactly)

Reply:
done